client/ayon_houdini/plugins/publish/collect_render_colorspace.py

```diff
@@ -32,14 +37,27 @@ def process(self, instance):
                            "Skipping collecting of render colorspace.")
             return
         aov_name = list(expected_files[0].keys())
-        render_products_data = colorspace.ARenderProduct(aov_name)
+        try:
+            render_products_data = ARenderProduct(aov_name)
+        except Exception as exc:
+            raise PublishError(
+                "Failed to get render products with colorspace.",
+                detail=f"{exc}"
+            )
```

Comment:
I really doubt that raising these errors is an improvement over _just letting it error_. The raised errors tells me basically nothing about _what_ caused it exactly and hence provides me as a developer nor as an artist the means to go solve it. Hence, the original error's exception must be (hopefully) clearer as to what is going on or failing.

I feel like our raised error is obfuscating the original error, no?

Also, note that you can do `raise PublishError("message") from exc` or `raise PublishError("message").with_traceback(exc.__traceback__)` although that's maybe a bit verbose. But I guess that's not raising an error in a way that looks nice in the Publisher UI.

What do we think @iLLiCiTiT @dee-ynput - are we obfuscating the errors and just making the code more complex or does this make sense? (Or should we instead still be changing how errors appear in the UI if NOT raised as `PublishError` so that a _regular_ error occurring in code still looks better). I'm a bit worried we'll now be getting tons of `PublishError("error occurred")` like errors added to the code just to provide the stack trace in the end to the UI instead of 'keeping it away from artists'.

Are we battling our own UI decisions here?

Take a look at the changes in this PR - and _what_ does it actually clarify to the raised errors, etc. whilst adding a lot more code for essentially raising the same error, but then as `PublishError` instead of the _actual_ error that occurred.

Reply:
For information, the current status of this PR aims to do the following: 
1. It raises publish errors instead of assert and other errors (from the caller functions in publish plugins).
2. It shows the original error as `detail` in publish error.
3. Enhance some error messages.

Personally I believe that `PublishError` should be used to guide the users about what could be wrong in their scene and how to fix it. e.g. `missing frames` -> `tells the users to ensure that all frames exist` 

Maybe I abused it a little bit to capture the potential errors that could happen from the called functions so that they show up nicely.